Add user-focused flows principle (#2318)
sourcegraph/handbook: content/strategy-goals/strategy/code-graph/index.md

- ✅ **Relevant**: In order to be useful, we need to be relevant. Our features will be the most accurate, relevant, and informative results possible.
- 🧠 **Educational**: For everyone to be able to code, we need to foster a culture of knowledge sharing. We are in a unique position to leverage search to increase code sharing and education, empowering communities of self-learners.

### User-focused workflows

Many of the features in Code Graph are used synergistically with each other to get the most out of the product; for example, you may discover a new shared component via some Search Notebook documentation, which leads you to running a query, perhaps setting up a new monitor via Insights, and then rolling out some upgrades via a Batch Change. These kinds of flows are exemplified by the common [use cases](../index.md#use-cases) we focus on.

> From an design standpoint, we aim to model user behavior with these cross-cutting workflows in mind first, and then empower individual teams to implement changes across Code Graph to make them reality. This requires coordinating with each other to make sure we stay on the same page, and the most common form of that is through [public RFCs](../../company-info-and-process/communication/rfcs.md) where we invite all of Code Graph engineering, design, and product, as well as customers and anyone else interested from Sourcegraph to participate. The situations we are trying to avoid are:

Author: ```suggestion
From an design standpoint, we aim to model user behavior with these cross-cutting workflows in mind first, and then empower individual teams to implement changes across Code Graph to make them reality. This requires coordinating with each other to make sure we stay on the same page, and the most common form of that is through [public RFCs](../../company-info-and-process/communication/rfcs/index.md) where we invite all of Code Graph engineering, design, and product, as well as customers and anyone else interested from Sourcegraph to participate. The situations we are trying to avoid are:
```

> From an design standpoint, we aim to model user behavior with these cross-cutting workflows in mind first, and then empower individual teams to implement changes across Code Graph to make them reality. This requires coordinating with each other to make sure we stay on the same page, and the most common form of that is through [public RFCs](../../company-info-and-process/communication/rfcs/index.md) where we invite all of Code Graph engineering, design, and product, as well as customers and anyone else interested from Sourcegraph to participate. The situations we are trying to avoid are:

Author: ```suggestion
From an design standpoint, we aim to model user behavior with these cross-cutting workflows in mind first, and then empower individual teams to implement changes across Code Graph to make them reality. This requires coordinating with each other to make sure we stay on the same page, and the most common form of that is through [public RFCs](../../../company-info-and-process/communication/rfcs/index.md) where we invite all of Code Graph engineering, design, and product, as well as customers and anyone else interested from Sourcegraph to participate. The situations we are trying to avoid are:
```

- ✅ **Relevant**: In order to be useful, we need to be relevant. Our features will be the most accurate, relevant, and informative results possible.
- 🧠 **Educational**: For everyone to be able to code, we need to foster a culture of knowledge sharing. We are in a unique position to leverage search to increase code sharing and education, empowering communities of self-learners.

### User-focused workflows

Many of the features in Code Graph are used synergistically with each other to get the most out of the product; for example, you may discover a new shared component via some Search Notebook documentation, which leads you to running a query, perhaps setting up a new monitor via Insights, and then rolling out some upgrades via a Batch Change. These kinds of flows are exemplified by the common [use cases](../index.md#use-cases) we focus on.

> From an design standpoint, we aim to model user behavior with these cross-cutting workflows in mind first, and then empower individual teams to implement changes across Code Graph to make them reality. This requires coordinating with each other to make sure we stay on the same page, and the most common form of that is through [public RFCs](../../../company-info-and-process/communication/rfcs/index.md) where we invite all of Code Graph engineering, design, and product, as well as customers and anyone else interested from Sourcegraph to participate. In general, whether workshopping designs, setting strategy, or collaborating on features, our goal is to be as transparent and collaborative as possible.

Author: ```suggestion
From an planning standpoint we aim to model user behavior with these cross-cutting workflows in mind first, and then empower individual teams to implement changes across Code Graph to make them reality. This requires coordinating with each other to make sure we stay on the same page, and the most common form of that is through [public RFCs](../../../company-info-and-process/communication/rfcs/index.md) where we invite all of Code Graph engineering, design, and product, as well as customers and anyone else interested from Sourcegraph to participate. In general, whether workshopping designs, setting strategy, or collaborating on features, our goal is to be as transparent and collaborative as possible.
```

The situations we are trying to avoid are:

- Good improvements don't get made because there isn't a way to build consensus around cross-cutting concerns.
- We lose track of the big picture for important user flows because we plan and create at the team's horizon.

> - Work gets stuck because it is blocked waiting for other teams with different priorities.

Reviewer: ```suggestion
- Work gets stuck because it is blocked waiting for other teams with different priorities.
- Changes are made without consulting the team owning the area being changed
```

Reviewer: This has both happened to me, and I've done this rather recently. Probably good to call it out here.

> Because the Cloud team occasionally focuses on what would otherwise be considered core features within Source Graph as they focus on growth and cloud, this principle extends to our collaboration with them.

Author: ```suggestion
Because the [Cloud org](../cloud/index.md) occasionally focuses on what would otherwise be considered core features within Source Graph as they focus on growth, this principle extends to our collaboration with them.
```
